ergo-merkle-tree/src/merkleproof.rs

```diff
@@ -52,21 +59,18 @@ impl LevelNode {
 )]
 #[derive(Clone, Debug)]
 pub struct MerkleProof {
-    pub(crate) leaf_data: [u8; 32],
+    pub(crate) leaf_data: Vec<u8>,
```

Comment:
There is one case where this may be required, which is NipopowAlgos::extension_proof_for, since ExtensionCandidate fields can be > 32 bytes. Looking at https://github.com/ergoplatform/ergo/blob/master/src/main/scala/org/ergoplatform/modifiers/history/extension/ExtensionCandidate.scala#L35, I don't see this function being used extensively, but I think it's best to keep the data size variable